### PR TITLE
Implement HTML Report Generation

### PR DESCRIPTION
**Pull request by StarKhan6368**
**What**: Implement HTML Report Generation

**Why**:

This pull request adds HTML report generation to improve the reviewing of test results. HTML interface provides better visibility and understanding of test outcomes compared to console output.

#### Feature

1. Generates comprehensive HTML reports of test outcomes, including coverage metrics.
2. Provides functionality to inspect and analyze individual request, response and duration details.
3. Includes a comprehensive summary of the tests, incorporating total duration, failures, errors, and more. 
4. Printable Test Coverage Report, customizability and more.

**Checklist**:

- [ ] Documentation added to the README.md OR link to PR on https://github.com/znsio/specmatic-documentation
- [ ] Tests
- [ ] Sonar Quality Gate

### CODE REVIEW

File: core/src/main/kotlin/io/specmatic/test/DataRecorder.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package io.specmatic.test
+
+import io.specmatic.core.log.HttpLogMessage
+
+object DataRecorder {
```
Comment (harikrishnan83) [inline core/src/main/kotlin/io/specmatic/test/DataRecorder.kt]: Please rename to `TestInteractionsLog`

File: core/src/main/kotlin/io/specmatic/test/DataRecorder.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    val testHttpLogMessages = mutableListOf<HttpLogMessage>()
+    val stubHttpLongMessages = mutableListOf<HttpLogMessage>()
```
Comment (harikrishnan83) [inline core/src/main/kotlin/io/specmatic/test/DataRecorder.kt]: I guess you meant to name this `stubHttpLogMessages`?

File: core/src/main/kotlin/io/specmatic/test/DataRecorder.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    fun addHttpLog(httpLogMessage: HttpLogMessage) {
+        if(httpLogMessage.scenario != null) {
```
Comment (harikrishnan83) [inline core/src/main/kotlin/io/specmatic/test/DataRecorder.kt]: This `if` block and null check (along with redundant else) we need to remove / restructure it

**Review by harikrishnan83 — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/io/specmatic/test/DataRecorder.kt (hunk 1 above)
Comment (StarKhan6368) [reply core/src/main/kotlin/io/specmatic/test/DataRecorder.kt]: Thanks, renamed it to `TestInteractionsLog`

File: core/src/main/kotlin/io/specmatic/test/DataRecorder.kt (hunk 2 above)
Comment (StarKhan6368) [reply core/src/main/kotlin/io/specmatic/test/DataRecorder.kt]: thanks, fixed the typo

File: core/src/main/kotlin/io/specmatic/test/DataRecorder.kt (hunk 3 above)
Comment (StarKhan6368) [reply core/src/main/kotlin/io/specmatic/test/DataRecorder.kt]: Thanks, simplified the logic and made it more readable.

**Review by harikrishnan83 — APPROVED** [review]
Looks good to me. Thanks.

**Pull request merged**